microsoft/PowerToys
# [KBM] Do not register low level hook if there're no remappings

## Summary of the Pull Request
Start/Stop hook when there's something to hook. Also, we must start hook from a thread that has a msg queue, so we use `PostThreadMessage` to achieve that by notifying the main thread.


## PR Checklist



## Detailed Description of the Pull Request / Additional comments


## Validation Steps Performed
1. start KBM w/o remappings, add one -> make sure it works
2. remove it, attach to KBME process from VS and pause it -> no key lagging means there's no hook.
3. start KBM w/ remappings -> verify that they work. 
4. remove them -> use the same VS trick as in `#2` to verify that there's no hook.
